### PR TITLE
test methods annotated with `@test`

### PR DESCRIPTION
**Pull request by staabm**
phpunit allows to mark methods with `@test`. these should not be marked as unused.

I think this works already but was just not cover by tests. here we are.

see https://docs.phpunit.de/en/9.5/writing-tests-for-phpunit.html#writing-tests-for-phpunit

### CODE REVIEW

File: src/Collectors/PublicClassMethodCollector.php [hunk 1]
```diff
@@ -49,12 +49,6 @@ public function processNode(Node $node, Scope $scope): ?array
             return null;
         }
 
-        // skip test methods
-        $classMethodName = $node->name->toString();
-        if (str_starts_with($classMethodName, 'test')) {
-            return null;
-        }
-
```
Comment (staabm) [inline src/Collectors/PublicClassMethodCollector.php]: this is already covered by the logic which skips everything which subclasses `PHPUnit\Framework\TestCase` (or requires a new test-case with a `test*`-method but does not subclass `PHPUnit\Framework\TestCase`)

Comment (TomasVotruba) [reply src/Collectors/PublicClassMethodCollector.php]: :+1: 

Could you add test for this one too?

Comment (staabm) [reply src/Collectors/PublicClassMethodCollector.php]: reverted and new tests added

**Comment by TomasVotruba** [comment]
Thank you 🙏

**Pull request merged**